spicy/toolchain/src/compiler/codegen/parser-builder.cc

```diff
@@ -493,12 +493,30 @@ struct ProductionVisitor
                 pushState(std::move(pstate));
             }
         }
+
+        auto __offsets = builder::member(state().self, "__offsets");
+        auto cur_offset = builder::memberCall(state().cur, "offset", {});
+
+        // Since the offset list is created empty make sure the element for the currrent field is present.
+        auto index = builder()->addTmp("index", builder::integer(field->index()));
+        auto resize = builder()->addWhile(builder::lowerEqual(builder::size(__offsets), index));
+        pushBuilder(resize);
+        builder()->addMemberCall(__offsets, "push_back", {builder::null()});
+        popBuilder();
```

Comment:
Shorter way to do this in this case:

```suggestion
        resize->addMemberCall(__offsets, "push_back", {builder::null()});
```

There's no need for `pushBuilder()` if we don't pass control elsewhere that doesn't know about the specific builder we want to use.